src/vm/checker/read_only/mod.rs

```diff
@@ -111,6 +111,21 @@ impl <'a, 'b> ReadOnlyChecker <'a, 'b> {
                       Ok(acc? && self.is_read_only(&argument)?) })
     }
 
+    fn are_tuple_values_read_only(&self, tuples: &[SymbolicExpression]) -> CheckResult<bool> {
```

Comment:
This method is missing a check that all entries in `tuples` are, in fact, tuples (assuming that this method is meant to operate on a slice of `SymbolicExpression`s that are assumed to be tuples?).  Please add a check in the `for` loop to verify this.